container/Operator_test.go

```diff
@@ -0,0 +1,392 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Genome Research Ltd.
+ *
+ * Author: Ashwini Chhipa <ac55@sanger.ac.uk>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ ******************************************************************************/
+
+package container
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+// fileMode is the mode of the temp file created for testing.
+const fileMode os.FileMode = 0600
+
+// mockInteractor represents a mock implementation of container.Interactor.
+type mockInteractor struct {
+	ContainerListFn       func() ([]*Container, error)
+	ContainerListInvoked  int
+	ContainerStatsFn      func(string) (*Stats, error)
+	ContainerStatsInvoked int
+	ContainerKillFn       func(string) error
+	ContainerKillInvoked  int
+}
+
+// ContainerList is a mock function which returns the list of containers.
+func (m *mockInteractor) ContainerList(ctx context.Context) ([]*Container, error) {
+	m.ContainerListInvoked++
+
+	return m.ContainerListFn()
+}
+
+// ContainerStats is a mock function which returns the mem and cpu stats of a
+// container given its ID.
+func (m *mockInteractor) ContainerStats(ctx context.Context,
+	containerID string) (*Stats, error) {
+	m.ContainerStatsInvoked++
+
+	return m.ContainerStatsFn(containerID)
+}
+
+// ContainerKill is a mock function which kills (removes the entry of) a
+// container given its ID.
+func (m *mockInteractor) ContainerKill(ctx context.Context, containerID string) error {
+	m.ContainerKillInvoked++
+
+	return m.ContainerKillFn(containerID)
+}
+
+// RemoveContainer removes the container from container list if found and
+// returns the remaining containers.
+func RemoveContainer(containerList []*Container, containerID string) []*Container {
```

Comment:
This currently "pollutes" the container package namespace, and it might get misused or misunderstood as something that works on real containers. Better to "hide" it by making it private or a method on MockInteractor. It might need to be public if other packages testing with MockInteractor are likely to need this.

Reply:
Made it private as it's not been used anywhere else, renamed it to removeContainerEntry.